libs/src/services/creatorNode/CreatorNodeSelection.js

```diff
@@ -0,0 +1,154 @@
+const axios = require('axios')
+const semver = require('semver')
+
+const ServiceSelection = require('../../service-selection/ServiceSelection')
+const { CREATOR_NODE_SERVICE_NAME } = require('./constants')
+
+class CreatorNodeSelection extends ServiceSelection {
+  constructor ({ creatorNode, numberOfNodes, ethContracts, whitelist, blacklist }) {
+    super({
+      getServices: async () => {
+        this.currentVersion = await ethContracts.getCurrentVersion(CREATOR_NODE_SERVICE_NAME)
+        const services = await this.ethContracts.getServiceProviderList(CREATOR_NODE_SERVICE_NAME)
+        return services.map(e => e.endpoint) // ? might need to map like services.map(e => e.endpoint)
+      },
+      whitelist
+    })
+    this.creatorNode = creatorNode
+    this.numberOfNodes = numberOfNodes
+    this.ethContracts = ethContracts
+    this.blacklist = blacklist
+
+    // List of valid past creator node versions registered on chain
+    this.validVersions = null
```

Comment:
wasn't too sure on how to use this variable / if we would need to integrate this eventually. currently not used